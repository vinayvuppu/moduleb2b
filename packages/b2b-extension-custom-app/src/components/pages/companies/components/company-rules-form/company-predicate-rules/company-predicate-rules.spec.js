import React from 'react';
import { shallow } from 'enzyme';
import CompanyPredicateRules from './company-predicate-rules';
import PredicateModalForm from '../predicate-modal-form';

jest.mock('react-intl', () => {
  const actual = jest.requireActual('react-intl');
  return {
    ...actual,
    useIntl: jest.fn(() => ({
      formatMessage: jest.fn(() => 'formatted message'),
    })),
  };
});

const createTestProps = props => ({
  formik: {
    setFieldValue: jest.fn(),
    values: {
      rules: [
        {
          name: 'name1',
          value: 'rule1',
        },
      ],
    },
  },
  isAuthorized: true,
  ...props,
});

describe('rendering', () => {
  let props;
  let wrapper;
  beforeEach(() => {
    props = createTestProps();
    wrapper = shallow(<CompanyPredicateRules {...props} />);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('interacting', () => {
  let props;
  let wrapper;

  describe('when press open modal form button', () => {
    beforeEach(() => {
      props = createTestProps();
      wrapper = shallow(<CompanyPredicateRules {...props} />);
      wrapper.find('SecondaryButton').simulate('click');
    });

    it('should render the "PredicateModalForm', () => {
      expect(wrapper).toRender(PredicateModalForm);
    });
  });
});
