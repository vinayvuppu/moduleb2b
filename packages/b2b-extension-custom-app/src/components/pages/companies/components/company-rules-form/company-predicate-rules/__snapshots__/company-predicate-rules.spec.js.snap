// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering should match snapshot 1`] = `
<Horizontal>
  <Stack
    alignItems="stretch"
    scale="s"
  >
    <Inline
      alignItems="center"
      justifyContent="flex-start"
      scale="s"
    >
      <SecondaryButton
        iconLeft={<PlusBoldIcon />}
        isDisabled={false}
        isToggleButton={false}
        label="formatted message"
        onClick={[Function]}
        theme="default"
        type="button"
      />
    </Inline>
    <Table
      columns={
        Array [
          Object {
            "key": "name",
            "label": "formatted message",
          },
          Object {
            "flexGrow": 1,
            "key": "value",
            "label": "formatted message",
          },
        ]
      }
      itemRenderer={[Function]}
      items={
        Array [
          Object {
            "name": "name1",
            "value": "rule1",
          },
        ]
      }
      onRowClick={[Function]}
      rowCount={1}
      shouldFillRemainingVerticalSpace={false}
    />
  </Stack>
</Horizontal>
`;
